src/main/java/mate/academy/springboot/datajpa/mapper/request/ProductRequestMapper.java

```diff
@@ -0,0 +1,25 @@
+package mate.academy.springboot.datajpa.mapper.request;
+
+import mate.academy.springboot.datajpa.mapper.DtoRequestMapper;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ProductRequestMapper implements DtoRequestMapper<ProductRequestDto, Product> {
+    private final CategoryService categoryService;
```

Comment:
Do not use service - id would be enough.

Reply:
method fromDto() should return object Product.  Object category  is the member of class Product.
which another method to  obtain category by Id?